AdminServer/appscale/admin/__init__.py

```diff
@@ -0,0 +1,404 @@
+""" A server that handles application deployments. """
+
+import argparse
+import datetime
+import logging
+import socket
+import sys
+import time
+import uuid
+
+from appscale.common import appscale_info
+from appscale.common.constants import HTTPCodes
+from appscale.common.constants import LOG_FORMAT
+from appscale.common.ua_client import UAClient
+from appscale.common.ua_client import UAException
+from appscale.common.unpackaged import APPSCALE_PYTHON_APPSERVER
+from tornado import gen
+from tornado.options import options
+from tornado import web
+from tornado.escape import json_decode
+from tornado.escape import json_encode
+from tornado.ioloop import IOLoop
+from . import utils
+from . import constants
+from .constants import (
+  CustomHTTPError,
+  MAX_DEPLOY_TIME,
+  OperationTimeout,
+  REDEPLOY_WAIT,
+  VALID_RUNTIMES
+)
+from .operations_cache import OperationsCache
+
+sys.path.append(APPSCALE_PYTHON_APPSERVER)
+from google.appengine.api.appcontroller_client import AppControllerException
+
+
+# The state of each operation.
+operations = OperationsCache()
+
+
+@gen.coroutine
+def wait_for_port_assignment(operation_id, deadline, acc):
+  """ Waits until port is assigned for version.
+
+  Args:
+    operation_id: A string specifying an operation ID.
+    deadline: A float containing a unix timestamp.
+    acc: An AppControllerClient.
+  Raises:
+    OperationTimeout if the deadline is exceeded.
```

Comment:
It would be unusual if operation was not in the cache since this function gets called immediately after the operation is added. However, I agree that that possibility should be dealt with. For now, I converted the KeyError to OperationTimeout since that seems more reasonable to catch by the caller.